src/decisionengine/framework/modules/de_logger.py

```diff
@@ -55,79 +51,70 @@ def set_logging(
     if dirname and not os.path.exists(dirname):
         os.makedirs(dirname)
 
-    logger.setLevel(getattr(logging, log_level.upper()))
-    if logger.handlers:
-        logger.debug("Reusing existing logging handlers")
+    delogger.setLevel(getattr(logging, log_level.upper()))
```

Comment:
Should this be prefixed with

```python
nonlocal delogger
```

So it operates on the item from line 17?  Otherwise I'm not sure what element it is targeting....

Reply:
If I add the nonlocal delogger to the get_logger function, I get the error "no binding for nonlocal 'delogger' found". Since delogger is defined at the top of the module in module scope, I don't think the nonlocal is necessary.

Reply:
Interesting, that is the opposite of what I'd expect since `delogger` isn't defined within the function.